src/libtexture/imagecache.cpp

```diff
@@ -3438,18 +3478,154 @@ namespace {
 
 
 
+void
+ImageCacheFile::udim_setup()
+{
+    // If it's a literal, existing file, always treat it as a regular
+    // texture, even if it has what looks like udim pattern markers.
+    if (Filesystem::exists(m_filename)) {
```

Comment:
Yep, nice, we have this same feature too, and it allows users to define explicit UDIM's as filenames transparently. Nicely done, nicely done.